.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build and Check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+permissions:
+  contents: read
+  security-events: write # upload Trivy Sarif results
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+        cache: false
+
+    - name: Tidy and verify modules
+      run: |
+        echo Tidying up modules
+        go mod tidy
+        echo Verifying modules
+        go mod verify
+
+    - name: Check formatting
+      run: test -z "$(go fmt $(go list ./...) | tee /dev/stderr)"
```

Comment:
This is not needed, the `golangci-lint` linter is already checking the formatting: https://github.com/rancher/kube-api-auth/blob/master/.golangci.json#L10

Reply:
I basically took all of the `scripts/validate`, to keep the differences between Drone and GHA down as much as possible. I.e. less chance to break some assumptions of the old flow.

I can make a note that this looks superfluous, and ask for a removal later, after the migration is done and properly working for a while.

Reply:
Comment added with commit 1c97df6fcea44baeb3d1fde45b761c49f65e5628